connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -497,6 +476,37 @@ static void validateHeaderConfigAction(String action) {
                     + "Expected one of %s", action, HEADER_ACTIONS));
         }
     }
+    private static class ListenersValidator implements ConfigDef.Validator {
+        @Override
+        public void ensureValid(String name, Object value) {
+            if (value == null) {
+                throw new ConfigException("Invalid value for listeners, at least one URL is expected, ex: http://localhost:8080,https://localhost:8443.");
+            }
+
+            if (!(value instanceof List)) {
+                throw new ConfigException("Invalid value type for listeners (expected list).");
+            }
+
+            List items = (List) value;
```

Comment:
Using generic types instead of raw types for collections is preferable (we can fix elsewhere in the file too)
```suggestion
            List<?> items = (List<?>) value;
```

Reply:
Fixed